azure/services/disks/client.go

```diff
@@ -30,19 +30,12 @@ import (
 )
 
 // client wraps go-sdk.
-type client interface {
-	DeleteAsync(context.Context, azure.ResourceSpecGetter) (azureautorest.FutureAPI, error)
-	Result(context.Context, azureautorest.FutureAPI, string) (interface{}, error)
-	IsDone(context.Context, azureautorest.FutureAPI) (bool, error)
-}
 
 // azureClient contains the Azure go-sdk Client.
 type azureClient struct {
 	disks compute.DisksClient
 }
 
-var _ client = (*azureClient)(nil)
-
 // newClient creates a new VM Client from subscription ID.
```

Comment:
```suggestion
// newClient creates a new disk Client from subscription ID.
 ```

not your PR but while you're here...